### PR TITLE
Add explicit symbol visibility macros to InstrProfData.inc

### PR DESCRIPTION
**Pull request by fsfod**
Add explicit symbol visibility macros to InstrProfData.inc

Annotating these symbols will fix missing symbols for InstrProfTest when
using shared library builds on windows with explicit visibility macros enabled.
Add a empty fallback definition for LLVM_ABI macro so the code works in compiler-rt.

This is part of the work to enable LLVM_BUILD_LLVM_DYLIB and plugins on window.

```
llvm\lld-link : error : undefined symbol: public: void ValueProfData::deserializeTo(InstrProfRecord&,  InstrProfSymtab*)
>>> referenced by unittests\ProfileData\InstrProfTest.cpp:1372 void ValueProfileReadWriteTest_value_prof_data_read_write_Test::TestBody()
```

### CODE REVIEW

File: compiler-rt/include/profile/InstrProfData.inc [hunk 1]
```diff
@@ -62,6 +62,15 @@
 #define INSTR_PROF_VISIBILITY
 #endif
 
+/* This is include is needed for symbol visibility macros used on
```
Comment (MaskRay) [inline compiler-rt/include/profile/InstrProfData.inc:65]: "This is include" => "This include"

That said, we usually add the comment to the use site.

Can you attach a full log of the build failure without this patch? It should be included to the patch description/commit message.

Comment (fsfod) [reply compiler-rt/include/profile/InstrProfData.inc:65]: so the comment should be by both the classes annotated with the macro from that header instead?

I should add the build failure will need a lot more code merged to trigger, but i wanted to give some context to this change.
This is the linker error:
```
ld.lld: error: undefined symbol: llvm::EnableABIBreakingChecks
>>> referenced by FindBugs.cpp
>>>               tools/bugpoint/CMakeFiles/bugpoint.dir/FindBugs.cpp.o:(.data._ZN4llvm29VerifyEnableABIBreakingChecksE+0x0)
>>> referenced by OptimizerDriver.cpp
>>>               tools/bugpoint/CMakeFiles/bugpoint.dir/OptimizerDriver.cpp.o:(.data._ZN4llvm29VerifyEnableABIBreakingChecksE+0x0)
>>> referenced by BugDriver.cpp
>>>               tools/bugpoint/CMakeFiles/bugpoint.dir/BugDriver.cpp.o:(llvm::VerifyEnableABIBreakingChecks)
```

Comment (fsfod) [reply compiler-rt/include/profile/InstrProfData.inc:65]: This is the linker error:
```
llvm\lld-link : error : undefined symbol: public: static class std::unique_ptr<struct llvm::ValueProfData, struct std::default_delete<struct llvm::ValueProfData>> __cdecl llvm::ValueProfData::serializeFrom(struct llvm::InstrProfRecord const &)
  >>> referenced by llvm\unittests\ProfileData\InstrProfTest.cpp:1368
  >>>               unittests/ProfileData/CMakeFiles/ProfileDataTests.dir/InstrProfTest.cpp.obj:(private: virtual void __cdecl `anonymous namespace'::ValueProfileReadWriteTest_value_prof_data_read_write_Test::TestBody(void))
  >>> referenced by llvm\unittests\ProfileData\InstrProfTest.cpp:1488
  >>>               unittests/ProfileData/CMakeFiles/ProfileDataTests.dir/InstrProfTest.cpp.obj:(private: virtual void __cdecl `anonymous namespace'::ValueProfileReadWriteTest_symtab_mapping_Test::TestBody(void))
  
llvm\lld-link : error : undefined symbol: public: void __cdecl llvm::ValueProfData::deserializeTo(struct llvm::InstrProfRecord &, class llvm::InstrProfSymtab *)
  >>> referenced by llvm\unittests\ProfileData\InstrProfTest.cpp:1372
  >>>               unittests/ProfileData/CMakeFiles/ProfileDataTests.dir/InstrProfTest.cpp.obj:(private: virtual void __cdecl `anonymous namespace'::ValueProfileReadWriteTest_value_prof_data_read_write_Test::TestBody(void))
  
  >>> referenced by llvm\unittests\ProfileData\InstrProfTest.cpp:1519
  
  >>>               unittests/ProfileData/CMakeFiles/ProfileDataTests.dir/InstrProfTest.cpp.obj:(private: virtual void __cdecl `anonymous namespace'::ValueProfileReadWriteTest_symtab_mapping_Test::TestBody(void))
  ```

**Comment by vgvassilev** [comment]
@MaskRay, do you think we can make progress with this PR given the answers from @fsfod?

**Comment by vgvassilev** [comment]
> @MaskRay, do you think we can make progress with this PR given the answers from @fsfod?

@MaskRay, any objections to move forward here?

**Comment by vgvassilev** [comment]
> > @MaskRay, do you think we can make progress with this PR given the answers from @fsfod?
> 
> @MaskRay, any objections to move forward here?

Let's move forward and rely on a post-commit review if necessary.

**Review by vgvassilev — APPROVED** [review]
LGTM! Let's move forward and reply on a post-commit reviews if necessary.

**Pull request merged**